tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -140,6 +219,74 @@ public string AsString(string defaultValue, Func<string, bool>? validator)
             return defaultValue.Result!;
         }
 
+        [return: NotNullIfNotNull(nameof(getDefaultValue))]
+        public T? GetAs<T>(Func<DefaultResult<T>>? getDefaultValue, Func<T, bool>? validator, Func<string, ParsingResult<T>> converter, string openTelemetryKey, Func<string, ParsingResult<T>> openTelemetryConverter)
+        {
+            var result = Source.GetAs<T>(Key, Telemetry, converter, validator, recordValue: true)
+                      ?? (FallbackKey1 is null ? null : Source.GetAs<T>(FallbackKey1, Telemetry, converter, validator, recordValue: true))
+                      ?? (FallbackKey2 is null ? null : Source.GetAs<T>(FallbackKey2, Telemetry, converter, validator, recordValue: true))
+                      ?? (FallbackKey3 is null ? null : Source.GetAs<T>(FallbackKey3, Telemetry, converter, validator, recordValue: true));
+
+            T? returnValue = default;
+            bool datadogConfigurationIsPresent = false;
+            if (result is { Result: { } value, IsValid: { } resultIsValid })
+            {
+                datadogConfigurationIsPresent = true;
+                if (resultIsValid)
+                {
+                    returnValue = value;
+                }
+            }
+            else
+            {
+                datadogConfigurationIsPresent = Source.IsPresent(Key)
```

Comment:
Do we need this branch? If `result` is null, doesn't that tell us everything we need to know?

Reply:
Yes we still need this branch, because `result` can be null if the configuration key is present but the value fails to be converted (e.g. by `converter` or by parsing to a number). In that case, we still want to acknowledge that the Datadog configuration was present so we don't apply OpenTelemetry configuration

Reply:
Hmmm, sounds like we should refactor those to return whether we _found_ a value as part of this then, rather than loading all the config for a second time 🤔 

We can improve it in a subsequent PR though, the important thing is this works

Reply:
Yeah I have an idea for improving the ConfigurationResult<T> class, but let's get this merged first 😆

TLDR; add a `IsPresent` field to `ConfigurationResult<T>` so differentiate it from the `null` value that's currently being returned in that scenario